cloud_functions/function.js

```diff
@@ -0,0 +1,72 @@
+const { project } = require('./config')
+const { SecretManagerServiceClient } = require('@google-cloud/secret-manager')
+const fetch = require('node-fetch')
+
+
+const flagCryptominer = async (message, context) => {
+  const data = JSON.parse(Buffer.from(message.data, 'base64').toString())
+  const token = await getSecret({ project, secretName: 'mixpanel-api' })
+  return fetchMixpanelV2('engage#profile-set', {
+    data: {
+      $token: token,
+      $distinct_id: `google:${data.googleSubjectId}`,
+      $set: {
+        'Is Cryptominer': true
+      }
+    }
+  })
+}
+
+// TODO: Share this code with the bard GAE app (will probably require a more sophisticated build/deploy process)
```

Comment:
Why weren't we able to share this code as-is?

Reply:
The directory structure and build scripts for both the GAE app and Cloud Function would need to change in order to make sure the right code went to the right deployments.